woocommerce/wc-admin
# Add country validation to subscription inclusion

This is a follow-up to [PR 7734](https://github.com/woocommerce/woocommerce-admin/pull/7734) to add a country validation.

The Subscription inclusion will be [visible only for `US` stores](https://github.com/woocommerce/woocommerce-admin/pull/7734#issuecomment-937440527) and this PR adds the changes necessary to accomplish that.

The requested changes here also required a small refactor to how we handle the product types. 

No changelog is necessary.

### Detailed test instructions:

0. Checkout this branch.
1. Deactivate and delete `WooCommerce Payments` if you have it installed.
2. Go to step one of the store profiler and select `France` (or any country other than the US) as the store `Country / Region`.
3. Go to step three of the store profiler (`Product Types`) and verify `Subscriptions` is shown as a paid extension (with a price chip).
5. Check `Subscriptions` and continue with the OBW.
6. Go back to the `Home` screen by pressing `Skip setup store details` in the step one of the store profiler. Check that the task item `Add Subscriptions to my store` is visible in the setup task list.
7. Press `Add my products` in the setup task list.
8. Select `Start with a template`. Verify that the option `Subscription product` is not visible in the popup.
9. Go to step one of the store profiler and select `US` as the store `Country / Region`.
10. Verify `Subscriptions` is shown as free (without a price chip). Also, verify that the text

```
The following extensions will be added to your site for free: WooCommerce Payments. An account is required to use this feature
```

is visible at the bottom when `WooCommerce Payments` is not installed.

![screenshot-one wordpress test-2021 09 30-14_12_58](https://user-images.githubusercontent.com/1314156/135506696-b7812f7e-437f-4d89-956a-b73248f70f6b.png)

11. Check `Subscriptions` and  press`Continue` and verify that the `WooCommerce Payments` plugin is installed and activated and it's not shown in the `Free features` list

![screenshot-one wordpress test-2021 09 30-14_32_20](https://user-images.githubusercontent.com/1314156/135506727-d8888f2b-3424-4cf5-a4bf-b67a14a198b6.png)

12. Verify that the `WooCommerce Payments` plugin is being shown in the `Free features` list when the store country is other than the `US`.

13. Go back to the `Home` screen by pressing `Skip setup store details` in step one of the store profiler. Check that the task item `Add Subscriptions to my store` is not visible in the setup task list. It should be visible if the store is from any country other than the US.

14. Press `Add my products` in the setup task list.
15. Select `Start with a template`. Verify that the option `Subscription product` is visible in the popup (verify that this is not being shown when the store is from a country other than the US)

![screenshot-one wordpress test-2021 09 30-14_35_22](https://user-images.githubusercontent.com/1314156/135506748-0b7bdce5-b006-47f9-9289-03ed26e4950c.png)


### UPDATE

15. Verify that [this error](https://github.com/woocommerce/woocommerce-admin/issues/7775) is not happening anymore.
